src/Tgstation.Server.Host/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -85,6 +89,30 @@ static class ServiceCollectionExtensions
 						sinkConfigurationAction?.Invoke(sinkConfiguration);
 					});
 
+				string elasticsearchEndpoint = tgsConfiguration.GetSection("Elasticsearch").GetSection("Host").Value;
+				string elasticsearchUser = tgsConfiguration.GetSection("Elasticsearch").GetSection("Username").Value;
+				string elasticsearchPassword = tgsConfiguration.GetSection("Elasticsearch").GetSection("Password").Value;
+				string raw_bool = tgsConfiguration.GetSection("Elasticsearch").GetSection("Enable").Value;
+				bool elasticsearchEnabled = false;
+
+				if (!string.IsNullOrEmpty(raw_bool))
+					elasticsearchEnabled = bool.Parse(raw_bool);
```

Comment:
This is the bit I am most set on "theres gotta be a better way"

Reply:
Yes, you should pass the `ElasticsearchConfiguration` class as a parameter here instead of `IConfiguration`. Include a `ElasticsearchConfiguration` property in `IPostSetupServices` so you can reference it from `Application`.